### PR TITLE
use real mean for importance ratio

### PR DESCRIPTION
**Pull request by samsja**

### CODE REVIEW

File: src/prime_rl/trainer/loss.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def grpo_loss_ratio(
     loss = _apply_mask(per_token_loss, loss_mask, max_tokens)
 
     with torch.no_grad():
```
Comment (mikasenghaas) [inline src/prime_rl/trainer/loss.py]: no need for the no_grad anymore?

File: src/prime_rl/trainer/loss.py [hunk 2]
```diff
@@ -141,7 +141,7 @@ def grpo_loss_ratio(
...
-        ratio_avg = _apply_mask(ratio, loss_mask, max_tokens)
+        ratio_avg = ratio.mean()
```
Comment (mikasenghaas) [inline src/prime_rl/trainer/loss.py]: why does `_apply_mask` not do the same?

File: src/prime_rl/trainer/loss.py [hunk 3]
```diff
@@ -141,7 +141,7 @@ def grpo_loss_ratio(
     loss = _apply_mask(per_token_loss, loss_mask, max_tokens)
 
     with torch.no_grad():
-        ratio_avg = _apply_mask(ratio, loss_mask, max_tokens)
+        ratio_avg = (loss_mask * ratio).sum() / loss_mask.sum()
```
Comment (mikasenghaas) [inline src/prime_rl/trainer/loss.py]: i still dont get why we would only do this for computing `ratio_avg` and not everywhere `_apply_mask` is used? won't we have the same issue with the loss?
